src/librustc_typeck/error_codes.rs

```diff
@@ -3793,6 +3793,40 @@ details.
 [issue #33685]: https://github.com/rust-lang/rust/issues/33685
 "##,
 
+E0592: r##"
+This error occurs when you defined methods or associated functions with same
+name.
+
+Erroneous code example:
+
+```compile_fail,E0592
+struct Foo;
+
+impl Foo {
+    fn bar() {} // previous definition here
+}
+
+impl Foo {
+    fn bar() {} // duplicate definition here
+}
+```
+
+A similar error is E0201. The difference is whether there is one declaration
+block or not. To avoid this error, you have to give them one name each.
```

Comment:
"To avoid this error, you have to give them one name each." -> I'm not sure this is a correct english wording but not being an english native, I'd prefer someone else to confirm this.

Reply:
Should we ask somebody directly? I'm not sure who is an English native...

Reply:
I think this is grammatically correct but not very illuminating... This should be:

```suggestion
block or not. To avoid this error, you must give each `fn` a unique name.
```

Reply:
Thanks @Centril ! I changed wording.